### ISSUE TITLE
[NEWS] Drafty produktów - wprowadziliśmy nowy filtr oraz pole w wyszukiwarce produktów / Products drafts - we have introduced a new field and filter in the product search resources

### ISSUE THREAD

**Issue by PrzemyslawLukanowski** [allegro/allegro-api#5954]
Drafty produktów to produkty, które zaczniemy tworzyć od 1 sierpnia po tym, gdy użytkownik strony WWW Allegro przekaże nam [sugestię zmiany parametrów](https://allegro.pl/dla-sprzedajacych/jak-zglosic-blad-w-produkcie-ZGZG89WkOU0), które identyfikują produkt. Draft produktu może od razu zostać wykorzystany przez sprzedającego, aby połączyć z nim ofertę - dzięki temu nie musi czekać na weryfikację poprawności danych, aby móc wystawić ofertę. Aktualnie drafty utworzymy **wyłącznie** na podstawie zgłoszeń z naszej strony internetowej.

Więcej informacji na temat tej zmiany znajdziesz na [stronie dla sprzedających](https://allegro.pl/dla-sprzedajacych/od-1-sierpnia-szybciej-wystawisz-oferte-ze-zmienionymi-parametrami-podstawowymi-produktu-wwmW8Ezq9cE).

**Jaki jest cykl życia draftu produktu?**
 
Jeśli sugestia zmiany danych produktu zostanie przez nas:

- zaakceptowana - zmiany z draftu uwzględnimy w produkcie głównym, do którego użytkownik zgłosił sugestię i uwzględnimy go w ofercie zamiast draftu, który usuniemy;
- odrzucona - nie uwzględnimy zmian w produkcie głównym, a produkt, który jest draftem, zostanie przez nas odłączony od oferty i usunięty. Wyjątkiem jest sytuacja, kiedy na podstawie sugestii wykryjemy inny wariant produktu - w takim przypadku draft stanie się pełnoprawnym produktem i pozostanie w ofercie;

**Jakie zmiany wprowadziliśmy?**

W zasobach [/sale/products](https://developer.allegro.pl/documentation#operation/getSaleProducts) i [/sale/products/{productId}](https://developer.allegro.pl/documentation#operation/getSaleProduct) dodaliśmy: 

- nowy parametr wyszukiwania **“includeDrafts”,** dzięki któremu określisz, czy zwrócony wynik ma uwzględniać drafty produktów (domyślna wartość to false);
- flagę “isDraft” w strukturze odpowiedzi, która określa, czy dany produkt jest draftem.

-----

Product drafts are products that we will start creating from August 1, after the user [reports a suggestion](https://allegro.pl/dla-sprzedajacych/jak-zglosic-blad-w-produkcie-ZGZG89WkOU0) on our Allegro website to change the parameters that identify the product. Product drafts can be used by sellers immediately to associate their offers. Thanks to that they do not have to wait for the suggestion verification to be able to list an offer. Currently, we will create drafts based only on reports from our website.

You will find more information about this change on the [For Sellers page](https://allegro.pl/dla-sprzedajacych/od-1-sierpnia-szybciej-wystawisz-oferte-ze-zmienionymi-parametrami-podstawowymi-produktu-wwmW8Ezq9cE).

**What is the life cycle of a product draft?**
 
If a suggestion to change product data will be:

- accepted - we will include changes from the draft in the main product, for which the suggestion has been made. The main product will be associated with the offer and the draft will be removed.
- declined - we will not include the changes in the main product and the main product will be removed from the offer and removed. The exception is when, based on the suggestion, we detect another variant of the product - in this case, the draft will become a full-fledged product and will be included in the offer.

**What changes have we made?**

In the [/sale/products](https://developer.allegro.pl/documentation#operation/getSaleProducts) and [/sale/products/{productId}](https://developer.allegro.pl/documentation#operation/getSaleProduct) resources we have added:

- new query parameter **"includeDrafts”,** owing to which you can indicate whether the results should include drafts of the products (default value is false);
- “isDraft” flag in the response structure, which define whether the given product is a draft or not;

**Comment by hahyes** [comment]
Czy w takiej ofercie draft zostanie zastąpiony zaakceptowanym produktem w polu `productId`?

**Comment by PrzemyslawLukanowski** [comment]
Tak, informację znajdziesz w newsie:

> (...) zaakceptowana - zmiany z draftu uwzględnimy w produkcie głównym, do którego użytkownik zgłosił sugestię i uwzględnimy go w ofercie zamiast draftu, który usuniemy;

**Comment by hahyes** [comment]
@PrzemyslawLukanowski Przepraszam, czytanie selektywne mi się chyba już w piątek włącza. ;)

Swoją drogą, czy planujecie dodać możliwość tworzenia draftów w API, czy to na zawsze pozostanie na portalu?

**Comment by PrzemyslawLukanowski** [comment]
Tak, planujemy, jednak na tę chwilę nie mogę podzielić się konkretną datą wdrożenia.

**Comment by FromAnyHole** [comment]
@PrzemyslawLukanowski - czyli dalej brniemy w absurdy...
Skoro od dzisiaj każda nowa oferta (w większości popularnych kategorii) ma mieć obowiązek powiązania z Produktem przy pierwotnej publikacji, to co stanie się z ofertą tymczasowo przypisaną do Draftu Produktu, który zostanie potem wykopany z systemu?? Oferta pozostanie goła czy może równocześnie zamknięta (bez Produktu) - bo przecież już NIE WOLNO...

**Comment by AureliuszBrussy** [comment]
@FromAnyHole w takim przypadku odepniemy ofertę od niezweryfikowanej wersji produktu. Aktualnie w takiej sytuacji pozostaje połączenie oferty z innym produktem lub utworzenie nowego produktu w Katalogu. Oferta nie zostanie zamknięta.

**Comment by FromAnyHole** [comment]
@PrzemyslawLukanowski / @AureliuszBrussy  - to są wszystko skutki tego totalnego szaleństwa z Produktami...

Nie było by potrzeby stawiania teraz rzeczy na głowie, gdyby zwyczajnie zachować (dla tych popularnych kategorii) docelowy wymóg [75%] połączenia ofert z katalogiem Produktów. Przecież i tak będą w serwisie oferty bez Produktu, wygonić Prezesa!!

**Comment by PrzemyslawLukanowski** [comment]
**Update:** Draft danego produktu może wykorzystać tylko ten sprzedający, który go utworzył. / Only the seller who created a given draft can use it.
